README.md

```diff
@@ -138,282 +141,386 @@ concurrently "npm:lint:*(!fix)"
-    to use when prefixing with `time`. Default: `yyyy-MM-dd HH:mm:ss.ZZZ`
-  - `additionalArguments`: list of additional arguments passed that will get replaced in each command. If not defined, no argument replacing will happen.
-
-> **Returns:** an object in the shape `{ result, commands }`.
->
-> - `result`: a `Promise` that resolves if the run was successful (according to `successCondition` option),
->   or rejects, containing an array of [`CloseEvent`](#CloseEvent), in the order that the commands terminated.
-> - `commands`: an array of all spawned [`Command`s](#Command).
-
-Example:
-
-```js
-const concurrently = require('concurrently');
-const { result } = concurrently(
-  [
-    'npm:watch-*',
-    { command: 'nodemon', name: 'server' },
-    { command: 'deploy', name: 'deploy', env: { PUBLIC_KEY: '...' } },
-    {
-      command: 'watch',
-      name: 'watch',
-      cwd: path.resolve(__dirname, 'scripts/watchers'),
-    },
-  ],
-  {
-    prefix: 'name',
-    killOthers: ['failure', 'success'],
-    restartTries: 3,
-    cwd: path.resolve(__dirname, 'scripts'),
-  },
-);
-result.then(success, failure);
-```
+<!-- INSERT GENERATED DOCS START -->
+
+### `CommandIdentifier` (type)
+
+Identifier for a command; if string, it's the command's name, if number, it's the index.
+
+### `CommandInfo` (interface)
+
+**Members:**
+
+- name (`string`) - Command's name.
+- command (`string`) - Which command line the command has.
+- env (`Record<string, unknown>`) - Which environment variables should the spawned process have.
+- cwd (`string`) - The current working directory of the process when spawned.
+- prefixColor (`string`) - Color to use on prefix of the command.
+- raw (`boolean`) - Output command in raw format.
+
+### `CloseEvent` (interface)
+
+**Members:**
+
+- command (`CommandInfo`)
+- index (`number`) - The command's index among all commands ran.
+- killed (`boolean`) - Whether the command exited because it was killed.
+- exitCode (`string | number`) - The exit code or signal for the command.
+- timings (`{ startDate: Date; endDate: Date; durationSeconds: number; }`)
+
+### `TimerEvent` (interface)
+
+**Members:**
+
+- startDate (`Date`)
+- endDate (`Date`)
+
+### `ChildProcess` (type)
+
+Subtype of NodeJS's child_process including only what's actually needed for a command to work.
+
+### `KillProcess` (type)
+
+Interface for a function that must kill the process with `pid`, optionally sending `signal` to it.
+
+### `SpawnCommand` (type)
+
+Interface for a function that spawns a command and returns its child process instance.
+
+### `Command` (class)
+
+### `CommandParser` (interface)
+
+A command parser encapsulates a specific logic for mapping `CommandInfo` objects
+into another `CommandInfo`.
+
+A prime example is turning an abstract `npm:foo` into `npm run foo`, but it could also turn
+the prefix color of a command brighter, or maybe even prefixing each command with `time(1)`.
+
+**Members:**
+
+- /\*\*
```

Comment:
[ts-readme](https://github.com/intuit/ts-readme) doesn't seem to be maintained and it doesn't seem to parse inner jsdoc like [this](https://github.com/open-cli-tools/concurrently/blob/main/src/command-parser/command-parser.ts#L11)

Reply:
How about using [typedoc](https://www.npmjs.com/package/typedoc-plugin-markdown)?

Reply:
I didn't knew this package,
It seems pretty nice and match our use-case here.

I'll update the PR with it.

Reply:
@SeiwonPark I've been quite busy lately.

I've implemented `typedoc` let me know what do you think before I can start working on automation.

Reply:
cc @gustavohenke @paescuj 

Pretty nice job! [Interfaces](https://github.com/Tarektouati/concurrently/blob/main/docs/interfaces/CloseEvent.md) are documented pretty nicely but not sure if [classes](https://github.com/Tarektouati/concurrently/blob/main/docs/classes/Command.md) are. (And some other fixes are required as well)

IMHO as @gustavohenke liked tables, current table-sorted [README](https://github.com/Tarektouati/concurrently/blob/main/README.md) style seems better. And there seems to be more configuration to do to automate the process. So how about just let maintainers/contributors manually update the document more readable in current stage?

I've recommended **typedoc** to be like some use cases (_e.g. [chatgpt-api](https://github.com/transitive-bullshit/chatgpt-api/blob/main/docs/readme.md), and [mermaid](https://github.com/mermaid-js/mermaid/blob/develop/docs/intro/index.md), ... and some other open sources_). **typedoc** could be an option but this might be too early for `concurrently` in this stage(because to use it like the above libraries, pretty much comments should be fixed and written to fit the **typedoc** configuration).

So maybe setting the issue(**automating documentation generation**) as `<Pending>` and keep @Tarektouati's current README style that contains table-sorted document (but yes update document manually..) would be enough. Or, rather implementing separate library _e.g. some-library-like-jsdoc-to-readme-parser_ would be a nice choice.

Reply:
Thanks ! 

Totally agree with @SeiwonPark, as I see the generation result, I'm not totally convinced. 
At least, it shows where codebase should be enhanced to have a better experience with it. 

Let me know if we keep "automating documentation generation" task and create other PR in that direction, or maybe just drop the related commits.

cc @gustavohenke @paescuj